### ISSUE TITLE
Runtime monitoring of ETH status

### ISSUE THREAD

**Issue by marcoaccame** [robotology/icub-firmware#51]
The EMS is not monitoring the status of its ETH switch in runtime.

There are methods of HAL which query via I2C some registers of the Micrel switch which contains diagnostics info. So fare we have information about link up/down and number of detected RX CRC errors.

This reading is not normally enabled because reading one register is an operation of several hundreds of micro-seconds and is incompatible with the control loop running at 1 Khz.

So far, the health of the ETH channel has been monitored with the sequence number mechanism: the UDP packet contains a sequence number which the receiver verifies. If it detects a mismatch then it issue an error about loss of a packet. 

However, it is wise to have the monitoring of the ETH status somehow running continuously so that one can better identify the cause of the loss of a UDP packet.

For instance, in https://github.com/robotology/icub-support/issues/83, it is experienced a single packet loss whose cause is difficult to understand. It is the only robot where this problem happens, hence it *may* be due to some HW or noise problem.

For this reason, I will implement a check of the ETH status which is compatible w/ the requirements of the control loop.

**Comment by marcoaccame** [comment]
The object EOtheETHmonitor is responsible to perform this check.

It starts a low priority worker task which executes periodically (by default every 100 ms) and performs all the I2C readings required to verify the health state of the ETH switch.

The control loop calls a _Tick() function at each iteration. This function just checks if there is new information produced by the worker task. If any, it  evaluates if any diagnostics message is to be sent to robotInterface and it sends it.

So far, EOtheETHmonitor monitors if the link goes up or down in each of the two P2 and P3 ports (and it issue an info message on any change of status). And for each active port it also checks for RX CRC errors (registers 0x0007 and 0x2007 of the micrel).

Other suitable registers can be checked which can be associated to loss of a packet, but at the moment their reading is supported by HAL. Such registers tells for instance is a TX or RX packet is dropped for lack of internal resources. It is not difficult however to change HAL and to add their reading in EOtheETHmonitor.

**Comment by marcoaccame** [comment]
I have tested a new version of FW which runs the EOtheETHmonitor.

The link up/down is correctly detected at deskbench. I did not test the CRC detection because it was correctly tested time ago when the HAL method was implemented.

I also have tested the application of the iCubGenova02 robot and the EOtheETHmonitor works without interfering with the control loop.

Hence I have produced a new FW version 2.22.
